cluster/gce/addons/apiserver-network-proxy/daemonset.yaml

```diff
@@ -0,0 +1,57 @@
+---
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  labels:
+    addonmanager.kubernetes.io/mode: Reconcile
+    k8s-app: proxy-agent
+  namespace: kube-system
+  name: proxy-agent
+spec:
+  selector:
+    matchLabels:
+      k8s-app: proxy-agent
+  updateStrategy:
+    type: RollingUpdate
+  template:
+    metadata:
+      labels:
+        k8s-app: proxy-agent
+      annotations:
+        scheduler.alpha.kubernetes.io/critical-pod: ''
+    spec:
+      priorityClassName: system-cluster-critical
+      # Necessary to reboot node
+      hostPID: true
+      volumes:
+        - name: pki
+          hostPath:
+            path: /etc/srv/kubernetes/pki
+      containers:
+        - image: gcr.io/google-containers/proxy-agent:v0.0.3
+          name: proxy-agent
+          command: ["/proxy-agent"]
+          args: [
+                  "--logtostderr=true",
+                  "--ca-cert=/etc/srv/kubernetes/pki/proxy-agent/ca.crt",
+                  "--agent-cert=/etc/srv/kubernetes/pki/proxy-agent/client.crt",
+                  "--agent-key=/etc/srv/kubernetes/pki/proxy-agent/client.key",
+                  "--proxy-server-host=__APISERVER_IP__",
```

Comment:
Why does this need to be passed in? Why not read it from the kubernetes service endpoints?